src/doppkit_toolbox/esri/toolboxes/doppkit_wrapper_toolbox.pyt

```diff
@@ -180,39 +151,42 @@ class FetchExport:
         app.pk = aoi_pk
         app.disable_ssl_verification = False
 
+
+        # Looking for error with invalid server URL
         try:
             sync(app, aoi_pk)
-        except:
+        except AttributeError:
             arcpy.AddError(
                 f"Cannot connect to server. Server URL {url} may not be valid."
             )
+        except RuntimeError:
+            arcpy.AddError(
+                f"Unauthorized request. GRiD Access Token or AOI Key: '{aoi_pk}' may not be valid."
+            )
+        else:
+            aprx = arcpy.mp.ArcGISProject("CURRENT")
+            active_map = aprx.activeMap
+            arcpy.env.addOutputsToMap = True
+            if active_map is None:
+                arcpy.AddMessage("Active Map is None")
+
+            elif named_parameters.add_to_map.value:
+
+                files_to_render = []
+                for root, dirs, files in os.walk(output_dir):
```

Comment:
For the future- this `os.walk` strategy will prove problematic. Any files that happen to already be in the designated output directory will be added to the map (if geospatial) or attempted to be added to the map. This should be tweaked so that the toolbox knows (from doppkit messages or otherwise) to only attempt to add things that just got downloaded from GRiD.